src/modules/navigator/geofence.cpp

```diff
@@ -118,6 +125,29 @@ bool Geofence::inside(const struct vehicle_global_position_s &global_position,
 
 bool Geofence::inside(double lat, double lon, float altitude)
 {
+	int32_t max_horizontal_distance = _param_max_hor_distance.get();
+	int32_t max_vertical_distance = _param_max_ver_distance.get();
+
+	if (max_horizontal_distance > 0 || max_vertical_distance > 0) {
+		bool updated = false;
+		orb_check(_home_position_sub, &updated);
+
+		if (updated) {
+			struct home_position_s home;
+			orb_copy(ORB_ID(home_position), _home_position_sub, &home);
+
+			float dist_xy = -1.0f;
```

Comment:
Ah, that doesn't make sense. The dangers of pushing untested code.

I'll store the home position in geofence once it's set, but should it check for an updated home position? Is it valid to change the home position later?